lib/libxdp/libxdp.c

```diff
@@ -947,9 +948,11 @@ struct xdp_program *xdp_program__open_file(const char *filename,
 		goto err;
 	}
 
-	err = reuse_bpf_maps(obj, filename, opts);
-	if (err)
-		goto err_close_obj;
+	if (reuse) {
+		err = reuse_bpf_maps(obj, filename, opts);
```

Comment:
Can't we detect that reuse is not possible in reuse_bpf_maps() instead of having the condition?

Reply:
I couldn't find any. The point in the execution where reuse_bpf_maps() gets called is at the part where the execution is the same for native and for hw mode. The arguments with which xdp_program__open_file()/reuse_bpf_maps() do not give any information about that. So maybe introduce xdp_program__open_file() that does not reuse?

Reply:
Hmm, yeah we may need a new open function specifically for that. Bugger.

What happens if you leave out this bit? HW mode fails completely if it's using maps, right?

Reply:
It will try to use maps with different ifindexes than the program, and fail.

Reply:
Right, so sorry for taking so long to circle back to this. I think maybe in this instance we should make use of the fact that the call to reuse_bpf_maps() is a stopgap solution until we can use the freplace reattach functionality in the kernel (at which point the call to reuse is no longer necessary as we can handle already-loaded programs).

So in other words, if you just remove the call to reuse_bpf_maps() entirely as part of this series, we can at least make progress on fixing the HW mode stuff :)